addons/payment_razorpay/models/payment_token.py

```diff
@@ -0,0 +1,39 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, _
+from odoo.addons.payment_razorpay import const
+
+
+class PaymentToken(models.Model):
+    _inherit = 'payment.token'
+
+    def _get_razorpay_limit_exceed_warning(self, amount):
+        """ Return the maximum amount that can be paid using this token.
+
+        :param int amount: The amount to be paid (in INR).
+        :return: Token limit exceed warning message if amount is higher then limit else False.
+        :rtype: str
+        """
+        self.ensure_one()
+
+        if not amount or self.provider_code != 'razorpay':
+            return False
+
+        Transaction = self.env['payment.transaction']
+        # Find first transaction from which this token was created.
+        primary_token_tx = Transaction.search(
+            [('token_id', '=', self.id), ('operation', 'not in', ['offline', 'online_token'])],
+            limit=1,
+        )
+        if primary_token_tx:
+            mandate_max_amount = primary_token_tx._get_mandate_max_amount()
+        else:
+            # Retrieve the payment method code for this token and get maximum amount based on that.
+            random_token_tx = Transaction.search([('token_id', '=', self.id)], limit=1)
+            pm_code = (random_token_tx.payment_method_id.primary_payment_method_id or random_token_tx.payment_method_id).code
+            mandate_max_amount = const.MANDATE_MAX_AMOUNT.get(pm_code, const.MANDATE_MAX_AMOUNT['card'])
```

Comment:
Could you please format it so that these lines stay under 100 chars? ^^

Reply:
Sure :100: